### ISSUE TITLE
Learning martial arts from book does not make you weary

### ISSUE THREAD

**Issue by Tairesh** [CleverRaven/Cataclysm-DDA#49176]
# Describe the bug
When you are training using the martial arts book it is Activity: None, so you don't get tired at all.

# Steps To Reproduce
1. Get some martial art book (i used "The Shotokan Karate Handbook")
2. Read it
3. Check that Activity displayed on sidebar says None

# Expected behavior
Training should be at least the same activity as fighting (Extreme level, I suppose)

# Screenshots
![image](https://user-images.githubusercontent.com/1105196/120825717-1938dd00-c573-11eb-89af-819965624df3.png)

# Versions and configuration
 - OS: Linux Mint 20
 - Game Version: fe42098
 - Graphics version: Tiles
 - Ingame language: English
 - Mods loaded: [ "dda" ]

# Additional context

**Comment by Anaglyphic** [comment]
Hi.

Sorry I clutter up everything. But dude it's even worse than that, you are considered 'resting' reading the book, and will _recover_ some weariness. You do expend stamina practicing, though. But it's over quickly. If you have the intelligence to read the book you'll be a trained-up badass in ~2h.

Martial arts books aren't regular skill books, this is separate from your skill and being proficient, you basically found a magic 80-page paperback from the '80s someone ordered out of the back of an issue of Black Belt magazine, and one reading ( _martial_art_learned_from = MA_MANUAL_ ) and one session of jumping around, embedding all those bad habits because no knowledgeable practitioner is there to correct you, and you're now on The Path, Geralt. Go kill monsters.

With Str 12 Int 11, I just read the Shotokan book, danced around the room a bit looking at drawings diagrams and hopefully pictures with text, and then mere 104 minutes later, _volia!_ I'm a self-taught practitioner of Shotokan karate! Not over months, not even a whole day. I didn't even have to stop to rest once, nor eat drink or sleep, nor take any damage, nor incur a status effect "aches_pains_training" -2 all stats for the next day, nothing. _(I wonder - if I found a working VHS or DVD player and nerd's library of old '70s kung fu movies, imagine what I could learn, it'd be like "The Matrix"!!)_

Stats: 13(12)/11(10)/11(10)/10(9)

Start time: 20:42:51
Start stam: |||||
Start wear: Light
Start Wtra: |||||

(read book)
Requires intelligence of 9 to easily read. You can learn Karate style from it. This fighting style is hard to learn. It wold be easier to master if you'd have skill expertise in unarmed combat. A training session with the book takes 10 minutes.

Time: 20:43:47

(read book)
Train Karate from manual:
1  Train once
0  Train until tired or success

(press 0)
```
Now reading The Shotokan Karate Handbook. Press . or 5 to stop early.
You try again. This training will finally pay off.
You try again. This training will finally pay off.
This martial art is not easy to grasp. You start training the moves from the beginning.
You train the moves according to the book, but can't get a grasp of the style, so you start from the beginning.
You try again. This training will finally pay off.
---> You're feeling a bit better rested from your exertions. <--
This martial art is not easy to grasp. You start training the moves from the beginning.
You train the moves according to the book, but can't get a grasp of the style, so you start from the beginning.
You decide to read the manual and train even more. In martial arts, patience leads to mastery.
You try again. This training will finally pay off.
You train the moves according to the book, but can't get a grasp of the style, so you start from the beginning.
This martial art is not easy to grasp. You start training the moves from the beginning.
You learn Karate.
Press _ to select martial arts style.
```
Ending time: 22:27:21
Ending stam: ||||.
Ending wear: Fresh
Ending Wtra: \

Elapsed time to learn Shotokan Karate: 01:44:30

This is why think we need to be mindful around these types of issues - sure you could increment some serious training weariness, hell I would enjoy it and think it would be fun to learn a martial art over many in-game days, even weeks, with a variable ever-so-slightly increasing effectiveness like proficiencies do, practicing every night after you make your camp or get back to your hidey hole with the day's loot and cook your meal, or first thing out of bed, fits right into immersion - but it would probably drive others crazy and they'd just cheat to make it "_like it used to be_". If you annoy everyone and they just bypass all the hard effort to code test and debug the additions? If you extreme exhaust -300% malus them for learning a martial arts and they just debug it away, what's the point? 
\- \- \- \-

**Comment by Tairesh** [comment]
Maybe all martial arts book should be moved to some mod? It means no sense when there is no possibility to learn how to drive a helicopter at all but you can learn karate in two hours of resting, I don't understand why these books in vanilla.

**Comment by Salty-Panda** [comment]
With how INT scales for them, Cthulhu or slime character can learn it in under a minute.
Imo, it could just give you proficiency, with various martial arts' abilities additionally locked behind certain progress of it.

So read the book, you get 0% Karate proficiency.
The Karate stance requires nothing.
Karate chop requires 3lvl melee and at least 30% Karate proficiency.
Proficiency levels up alongside combat skills (following the same boosts) when you have it selected.

**Comment by actual-nh** [comment]
> Maybe all martial arts book should be moved to some mod? It means no sense when there is no possibility to learn how to drive a helicopter at all but you can learn karate in two hours of resting, I don't understand why these books in vanilla.

Are there NPCs that can teach each of the ones findable in books?

**Comment by Tairesh** [comment]
> > Maybe all martial arts book should be moved to some mod? It means no sense when there is no possibility to learn how to drive a helicopter at all but you can learn karate in two hours of resting, I don't understand why these books in vanilla.
> 
> Are there NPCs that can teach each of the ones findable in books?

No, but there is a very nice mod called [Wandering Masters](https://github.com/El-Jekozo/Wandering_Masters), it's currently WIP, but I think it would be more meaningful if there will be NPCs-Teachers or martial arts instead of self-education books

**Comment by eJ0opb** [comment]
> Imo, it could just give you proficiency, with various martial arts' abilities additionally locked behind certain progress of it.

This is less silly than the current system, but still not really how it works in real life - you can't learn fighting from a book; as Anaglyphic points out, it's probably actually one of the worst ways to get started because of all the bad habits you'll teach yourself.  Reference materials are actually more useful for instructors than for students - just because you know how to do something well doesn't mean you know how to teach someone else how to do it, and manuals provide explanations for techniques.

In my opinion the way it should probably work is
* The only ways to acquire martial arts proficiency are through starting traits and NPC interaction
* Start with a "Principles of" proficiency that doesn't give you any abilities and requires NPC training to progress beyond
* Manuals in possession of the *trainer* (not the trainee) may increase the rate of learning
* Once the "Principles of" proficiency has been learned, further proficiency is acquired through combat and abilities are unlocked by proficiencies

Actually the "unlock abilities" aspect of this isn't particularly realistic since real-life combat isn't really about "special attacks".  But changing that is way out of scope for this discussion.  Also - I make no claim as to whether or not this system is fun.

**Comment by NancokPS2** [comment]
> Maybe all martial arts book should be moved to some mod? It means no sense when there is no possibility to learn how to drive a helicopter at all but you can learn karate in two hours of resting, I don't understand why these books in vanilla.

This is from the times where you could self install CBMs, there just wasn't time to do anything more complex, MA def need a rework and since the dev in charge of them asked in reddit about feedback i imagine he's working on them

**Comment by sulyi** [comment]
TL;DR;
But that does not stop me adding to it.

Unless I missed something, it is possible to set `activity_level` key on (individual or abstract) books in json. 
But it has no effect on reading a book:
https://github.com/CleverRaven/Cataclysm-DDA/blob/53398d1e8576dc381e244d1ae2928fced4a642b2/src/activity_actor.cpp#L1440-L1469

Hoping this is useful.
